### DISCUSSION TITLE
渲染入口，控制渲染

### DISCUSSION THREAD

**Discussion by jmy2022** [umijs/umi#10747] [Q&A]
要加重一个功能库，然后才能开始UI渲染，该放在什么地方呢

**Comment by fz6m** [comment]
全部渲染开始前在 `src/app.ts` > [`render`](https://umijs.org/docs/api/runtime-config#renderoldrender-function) 方法里。

如果是一部分，可以放到全局 layout `src/layouts/index.tsx` 里，没准备好你的逻辑前渲染 loading 就行。

**Comment by jmy2022** [comment]
umi app.ts 里怎么定义render

**Comment by beilo** [comment]
在app.ts里,将root包裹起来.
这样可以满足嘛?
```
export function rootContainer(root) {
  return <Provider store={store}>{root}</Provider>;
}
```

**Comment by jmy2022** [comment]
谢谢两位，解决啦
